subsys/bluetooth/controller/ll_sw/ull_central_iso.c

```diff
@@ -725,32 +743,166 @@ uint16_t ull_central_iso_cis_offset_get(uint16_t cis_handle, uint32_t *cis_offse
 
+	cis->central.instant = ull_conn_event_counter(conn) + 3U;
+	*conn_event_count = cis->central.instant;
+
 	/* Provide CIS offset range
 	 * CIS_Offset_Max < (connInterval - (CIG_Sync_Delay + T_MSS))
 	 */
+	cig = cis->group;
 	*cis_offset_max = (conn->lll.interval * CONN_INT_UNIT_US) -
 			  cig->sync_delay;
 
 	if (IS_ENABLED(CONFIG_BT_CTLR_JIT_SCHEDULING)) {
 		*cis_offset_min = MAX(400, EVENT_OVERHEAD_CIS_SETUP_US);
+
+		return 0;
+	}
+
+#if (CONFIG_BT_CTLR_CENTRAL_SPACING == 0)
+	if (cig->started) {
+		cis_offset_get(cis);
 	} else {
-#if (CONFIG_BT_CTLR_CENTRAL_SPACING > 0)
-		*cis_offset_min = HAL_TICKER_TICKS_TO_US(conn->ull.ticks_slot) +
-				  (EVENT_TICKER_RES_MARGIN_US << 1U) +
-				  cig->sync_delay - cis->sync_delay;
-#else /* !CONFIG_BT_CTLR_CENTRAL_SPACING */
-		LL_ASSERT(false);
-#endif /* !CONFIG_BT_CTLR_CENTRAL_SPACING */
+		cig_offset_get(cis);
 	}
 
-	cis->central.instant = ull_conn_event_counter(conn) + 3;
+	return -EBUSY;
+#endif /* CONFIG_BT_CTLR_CENTRAL_SPACING  != 0 */
+
+	*cis_offset_min = HAL_TICKER_TICKS_TO_US(conn->ull.ticks_slot) +
+			  (EVENT_TICKER_RES_MARGIN_US << 1U) +
+			  cig->sync_delay - cis->sync_delay;
+
+	return 0;
+}
+
+#if (CONFIG_BT_CTLR_CENTRAL_SPACING == 0)
+static void cig_offset_get(struct ll_conn_iso_stream *cis)
+{
+	static memq_link_t link;
+	static struct mayfly mfy = {0, 0, &link, NULL, mfy_cig_offset_get};
+	uint32_t ret;
+
+	mfy.param = cis;
+	ret = mayfly_enqueue(TICKER_USER_ID_ULL_HIGH, TICKER_USER_ID_ULL_LOW, 1,
+			     &mfy);
+	LL_ASSERT(!ret);
+}
+
+static void mfy_cig_offset_get(void *param)
+{
+	struct ll_conn_iso_stream *cis;
+	struct ll_conn_iso_group *cig;
+	uint32_t ticks_to_expire;
+	struct ll_conn *conn;
+	uint32_t offset_us;
+	int err;
 
-	return cis->central.instant;
+	cis = param;
+	cig = cis->group;
+
+	err = ull_sched_conn_iso_free_offset_get(cig->ull.ticks_slot,
+						 &ticks_to_expire);
+	LL_ASSERT(!err);
+
+	offset_us = HAL_TICKER_TICKS_TO_US(ticks_to_expire) +
+		    (EVENT_TICKER_RES_MARGIN_US << 1U) +
+		    cig->sync_delay - cis->sync_delay;
+
+	conn = ll_conn_get(cis->lll.acl_handle);
+
+	ull_cp_cc_offset_calc_reply(conn, offset_us);
+}
+
+static void cis_offset_get(struct ll_conn_iso_stream *cis)
+{
+	static memq_link_t link;
+	static struct mayfly mfy = {0, 0, &link, NULL, mfy_cis_offset_get};
+	uint32_t ret;
+
+	mfy.param = cis;
+	ret = mayfly_enqueue(TICKER_USER_ID_ULL_HIGH, TICKER_USER_ID_ULL_LOW, 1,
+			     &mfy);
+	LL_ASSERT(!ret);
+}
+
+static void mfy_cis_offset_get(void *param)
+{
+	struct ll_conn_iso_stream *cis;
+	struct ll_conn_iso_group *cig;
+	uint32_t ticks_to_expire;
+	uint32_t ticks_current;
+	struct ll_conn *conn;
+	uint32_t remainder;
+	uint32_t offset_us;
+	uint8_t ticker_id;
+	uint16_t lazy;
+	uint8_t retry;
+	uint8_t id;
+
+	cis = param;
+	cig = cis->group;
+	ticker_id = TICKER_ID_CONN_ISO_BASE + ll_conn_iso_group_handle_get(cig);
+
+	id = TICKER_NULL;
+	ticks_to_expire = 0U;
+	ticks_current = 0U;
+	retry = 4U;
```

Comment:
What's the rationale behind retrying 4 times? Could you add that as a comment for this value?

Reply:
Added comment.